### PR TITLE
Rewrite python_bindings/apps

### PR DESCRIPTION
**Pull request by steven-johnson**
The apps were pretty old; none used Generators (though they really should have); most of them had very limited scheduling options. I updated all of them based on the equivalent C++ app, moved them to be Generator based (and self-contained instead of mixed in with the tests), and nuked the `erode` app entirely since it was of dubious use and had no C++ counterpart. Also formatted all of them with Black (https://github.com/psf/black), which is one of most popular Python formatters. The goal here is to ensure that the apps here are good examples of "best practices" for Halide Python use as of Halide 15.

### CODE REVIEW

File: python_bindings/test/apps/bilateral_grid_generator.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            pass
+        elif g.target().has_gpu_feature():
+            # 0.50ms on an RTX 2060
+
+            xi, yi, zi = _vars("xi", "yi", "zi")
+
+            # Schedule blurz in 8x8 tiles. This is a tile in
+            # grid-space, which means it represents something like
+            # 64x64 pixels in the input (if s_sigma is 8).
+            (blurz.compute_root().reorder(c, z, x, y).gpu_tile(x, y, xi, yi, 8, 8))
```
Comment (abadams) [inline python_bindings/test/apps/bilateral_grid_generator.py:129]: Is it worth it to use parens for the one-liner schedules, or is it just confusing?

File: python_bindings/test/apps/blur_app.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+"""
+Bilateral histogram.
```
Comment (abadams) [inline python_bindings/test/apps/blur_app.py:2]: Wrong comment?

File: python_bindings/test/apps/bilateral_grid_generator.py (hunk 1 above)
Comment (steven-johnson) [reply python_bindings/test/apps/bilateral_grid_generator.py:129]: Not sure. I basically wrapped all of the multi-schedule bits in parens and let Black do its thing, and, well, this is what it does. Not sure I like it but apparently that's the way the Python ecosystem seems to be moving. (Google is moving towards `pyink` internally, which is basically a fork of Black with a handful of minor changes to minimize churn.)

File: python_bindings/test/apps/local_laplacian_generator.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
+"""
+Fast image interpolation using a pyramid.
```
Comment (abadams) [inline python_bindings/test/apps/local_laplacian_generator.py:2]: This comment looks like it's from the interpolate app

File: python_bindings/test/apps/blur_app.py (hunk 2 above)
Comment (steven-johnson) [reply python_bindings/test/apps/blur_app.py:2]: oopsie, fixing

File: python_bindings/test/apps/local_laplacian_generator.py (hunk 3 above)
Comment (steven-johnson) [reply python_bindings/test/apps/local_laplacian_generator.py:2]: oopsie, fixing

File: python_bindings/test/apps/bilateral_grid_generator.py (hunk 1 above)
Comment (steven-johnson) [reply python_bindings/test/apps/bilateral_grid_generator.py]: I decided to remove all the parens from the cases that fit on a single line.

**Comment by steven-johnson** [comment]
Monday Morning Review Ping

**Comment by abadams** [comment]
Are these apps run in CI? I can't figure out which builder does it.

**Comment by steven-johnson** [comment]
> Are these apps run in CI? I can't figure out which builder does it.

Yes, anything that runs the `python` tests exercises these.

**Review by abadams — APPROVED** [review]

**Pull request merged**